nomium/shares-logger/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+pub mod config;
+pub mod errors;
+pub mod models;
+pub mod services;
+pub mod storage;
+pub mod traits;
+pub mod worker_name_store;
+
+use crate::config::SETTINGS;
+use crate::models::BlockFound;
+use crate::models::ShareLog;
+use crate::storage::clickhouse::ClickhouseBlockStorage;
+use crate::storage::clickhouse::ClickhouseStorage;
+use crate::traits::ShareStorage;
+use lazy_static::lazy_static;
+use log::info;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::sync::Arc;
+use std::time::Instant;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tokio::sync::{
+    mpsc::{self, error::TrySendError},
+    Mutex,
+};
+use tokio::time::Duration;
+
+use nomium_prometheus::{
+    SHALOG_BACKUP_CHANNEL_CURRENT, SHALOG_BACKUP_CHANNEL_SHARES_TOTAL,
+    SHALOG_BACKUP_STORE_FAILED_TOTAL, SHALOG_BACKUP_TRY_STORED_TOTAL,
+    SHALOG_PRIMARY_CHANNEL_CURRENT, SHALOG_PRIMARY_CHANNEL_SHARES_TOTAL,
+    SHALOG_PRIMARY_STORE_FAILED_TOTAL, SHALOG_PRIMARY_TRY_STORED_TOTAL,
+};
+
```

Comment:
Ensure all required Prometheus metrics are correctly initialized and documented.